libs/src/api/challenge.js

```diff
@@ -0,0 +1,310 @@
+const axios = require('axios')
+const { Base, Services } = require('./base')
+const { sampleSize } = require('lodash')
+const BN = require('bn.js')
+
+const GetAttestationError = Object.freeze({
+  CHALLENGE_INCOMPLETE: 'CHALLENGE_INCOMPLETE',
+  ALREADY_DISBURSED: 'ALREADY_DISBURSED',
+  INVALID_ORACLE: 'INVALID_ORACLE',
+  MISSING_CHALLENGES: 'MISSING_CHALLENGES',
+  INVALID_INPUT: 'INVALID_INPUT',
+  HCAPTCHA: 'HCAPTCHA',
+  COGNITO_FLOW: 'COGNITO_FLOW',
+  UNKNOWN_ERROR: 'UNKNOWN_ERROR'
+})
+
+const AttestationPhases = Object.freeze({
+  AGGREGATE_ATTESTATIONS: 'AGGREGATE_ATTESTATIONS',
+  SUBMIT_ATTESTATIONS: 'SUBMIT_ATTESTATIONS',
+  EVALUATE_ATTESTATIONS: 'EVALUATE_ATTESTATIONS'
+})
+
+const AAO_REQUEST_TIMEOUT_MS = 15 * 1000
+const WRAPPED_AUDIO_PRECISION = 10 ** 9
+
+/**
+ * @typedef {import("../services/solanaWeb3Manager/rewards.js").AttestationMeta} AttestationMeta
+ */
+
+class Challenge extends Base {
+  /**
+   *
+   * Top level method to aggregate attestations, submit them to RewardsManager, and evalute the result.
+   *
+   * @param {{
+   *   challengeId: string,
+   *   encodedUserId: string,
+   *   handle: string,
+   *   recipientEthAddress: string,
+   *   specifier: string,
+   *   oracleEthAddress: string,
+   *   amount: number,
+   *   quorumSize: number,
+   *   AAOEndpoint: string
+   * }} {
+   *   challengeId,
+   *   encodedUserId,
+   *   handle,
+   *   recipientEthAddress,
+   *   specifier,
+   *   oracleEthAddress,
+   *   amount,
+   *   quorumSize,
+   *   AAOEndpoint
+   *   }
+   * @memberof Challenge
+   */
+  async submitAndEvaluate ({
+    challengeId, encodedUserId, handle, recipientEthAddress, specifier, oracleEthAddress, amount, quorumSize, AAOEndpoint
+  }) {
+    let phase
+    try {
+      phase = AttestationPhases.AGGREGATE_ATTESTATIONS
+      const { discoveryNodeAttestations, aaoAttestation, error } = await this.aggregateAttestations({
+        challengeId, encodedUserId, handle, specifier, oracleEthAddress, amount, quorumSize, AAOEndpoint
+      })
+
+      if (error) {
+        return { success: false, error }
+      }
+
+      const fullTokenAmount = new BN(amount * WRAPPED_AUDIO_PRECISION)
```

Comment:
what if amount is a fraction?

i think we may run into some very weird precision issues.
generally, amount _shouldn't_ be a fraction here, but you never know. worth an assertion?